src/btree/bt_walk.c

```diff
@@ -624,3 +625,235 @@ __wti_tree_walk_skip(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *skipleaf
 
     return (0);
 }
+
+/*
+ * __wt_page_npos --
+ *     Get the page's normalized position in the tree. - \c start is a position within the leaf
+ *     page: - 0 .. 1. If \c path_str_offset is set, make a string representation of the path.
+ */
+double
+__wt_page_npos(WT_SESSION_IMPL *session, WT_REF *ref, double start, char *path_str,
+  size_t *path_str_offset, size_t path_str_sz_max)
+{
+    WT_PAGE_INDEX *pindex;
+    double npos;
+    uint32_t entries, slot;
+    int unused = 1; /* WT_UNUSED(snprintf) is screwed in GCC */
+
+    npos = start;
+    if (path_str)
+        *path_str_offset = 0;
+
+    WT_ENTER_PAGE_INDEX(session);
+    while (!__wt_ref_is_root(ref)) {
+        slot = UINT32_MAX; /* We get this invalid value in case of error */
+        __ref_index_slot(session, ref, &pindex, &slot);
+        entries = pindex->entries;
+        if (slot < entries) /* This checks for both UINT32_MAX and a potentially invalid number */
```

Comment:
Per the comment
> Alternatively, could assign it any "reasonable" estimate like 0.5 or 0 / 1 depending on the walk direction
 
I'd be happy to use 0.5

Reply:
Since this situation is currently imaginary, does it make any sense to add a branch which will never be executed? It's just a precaution like "assert".